### PR TITLE
Accomodate ess in new CI

### PR DESCRIPTION
**Pull request by nvaytet**
- Added a `prefix` parameter:
  - in the case of a project that builds a cpp library (scipp, scippneutron), the prefix is just an install directory for Cmake.
  - in the case of a noarch package, it is simply the path to the `src` directory, so that we can run the python tests and build the docs
  - the `prefix` parameter does not affect package builds that use `conda-build` as it is already doing what we need.

- the test to see if the `build_cpp.py` file exists has been removed, instead we are now just testing whether the package is noarch or not

- we do not want to deal with `c(l)cache` for noarch packages

### CODE REVIEW

File: build.yml [hunk 1]
```diff
@@ -47,12 +52,10 @@ jobs:
       - ${{ if contains(parameters.image, 'windows') }}:
         - bash: |
             echo "##vso[task.prependpath]$CONDA\Scripts"
-            echo "##vso[task.setvariable variable=scipp_install_prefix]$CONDA\scipp_install"
```
Comment (SimonHeybrock) [inline build.yml:50]: Can we avoid this bit entirely be making a cmake preset defining the install dir?

Comment (nvaytet) [reply build.yml:50]: The pipeline needs to know the location to be able to set the `PYTHONPATH` further down, so i'm not sure how you would accommodate that?

Comment (SimonHeybrock) [reply build.yml:50]: I guess, unless you want to hard-code it to `install` within the repo?

Comment (SimonHeybrock) [reply build.yml:50]: Maybe the solution would be to change our cmake setup so Python tests can be run via cmake? Basically CI might just do:

```sh
cmake --preset <preset>
cmake --build --preset <preset>
cmake --test --preset <preset>
```

where `<preset>` can be substituted from Azure with the correct OS?

File: CMakeLists.txt [hunk 2]
```diff
@@ -4,32 +4,7 @@
...
-      CACHE PATH "Path to installed clcache executable directory")
-  if(EXISTS ${CLCACHE_PATH})
-    set(CMAKE_VS_GLOBALS "CLToolExe=clcache.exe" "CLToolPath=${CLCACHE_PATH}"
-                         "TrackFileAccess=false")
-    message(STATUS "build will attempt to use clcache")
-  else(EXISTS ${CLCACHE_PATH})
-    message(STATUS "CLCACHE_PATH not specified. Not enabled.")
-    unset(CMAKE_VS_GLOBALS) # Clear all vs globals
-  endif(EXISTS ${CLCACHE_PATH})
-endif(MSVC)
```
Comment (SimonHeybrock) [inline CMakeLists.txt:25]: Was this bit meant to test whether the clcache setup on Azure works?

Comment (SimonHeybrock) [reply CMakeLists.txt:25]: Ignore this, I just saw you moved it.

**Review by SimonHeybrock — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (nvaytet) [reply CMakeLists.txt:25]: I honestly don't remember...

Comment (SimonHeybrock) [reply CMakeLists.txt:25]: Probably because I wrote it :D

**Pull request merged**